Modules/MFT/src/runDigitsRootFileReaderMFT.cxx

```diff
@@ -0,0 +1,144 @@
+#include "ITSMFTBase/Digit.h"
+#include "DataFormatsITSMFT/ROFRecord.h"
+
+using namespace o2;
+using namespace o2::framework;
+using namespace o2::itsmft;
+
+
+class DigitsRootFileReaderMFT : public o2::framework::Task
+{
+public:
+  //_________________________________________________________________________________________________
+  void init(framework::InitContext& ic)
+  {
+    LOG(INFO) << " In runDigitsRootFileReaderMFT::init ... entering ";
+
+    // open the input file
+    auto filename = ic.options().get<std::string>("mft-digit-infile");
+    mFile = std::make_unique<TFile>(filename.c_str(), "OLD");
+    if (!mFile->IsOpen()) {
+      LOG(ERROR) << "runDigitsRootFileReaderMFT::init. Cannot open the file: " << filename.c_str();
+      ic.services().get<ControlService>().readyToQuit(QuitRequest::All);
+      return;
+    }
+
+  }
+
+  //_________________________________________________________________________________________________
+
+  void run(framework::ProcessingContext& pc)
+  {
+    // get vector of ROF
+    std::unique_ptr<TTree> treeROF((TTree*)mFile->Get("MFTDigitROF"));
+    std::vector<o2::itsmft::ROFRecord> rofs, *profs = &rofs;
+    treeROF->SetBranchAddress("MFTDigitROF", &profs);
+    treeROF->GetEntry(0);
+   
+    // Check if there is a new ROF
+    auto nROFs = rofs.size();
+    if (currentROF >= nROFs) {
+      ///if (currentROF >= 50) {    
+      LOG(INFO) << " runDigitsRootFileReaderMFT::run. End of file reached";
+      pc.services().get<ControlService>().readyToQuit(QuitRequest::All);
+      return;
+    }
+    // prepare the rof output
+    std::vector<o2::itsmft::ROFRecord> oneROFvec;
+    std::copy(rofs.begin()+currentROF, rofs.begin()+currentROF+1, std::back_inserter(oneROFvec));
+    // get the current ROF
+    auto& rof = rofs[currentROF];
+    currentROF++;
+      
+    // get the digits in current ROF
+    // --> get digit branch
+    std::unique_ptr<TTree> treeDig((TTree*)mFile->Get("o2sim"));
+    std::vector<o2::itsmft::Digit> digits, *pdigits = &digits;
+    treeDig->SetBranchAddress("MFTDigit", &pdigits);
+    treeDig->GetEntry(0);
+    // --> find the ranges
+    int index = rof.getROFEntry().getIndex(); // first digit position
+    int nDigitsInROF = rof.getNROFEntries(); // number of digits
+    int lastIndex = index+nDigitsInROF;
+     // --> fill in the corresponding digits
+    std::vector<o2::itsmft::Digit> DigitsInROF;
+    std::copy(digits.begin()+index, digits.begin()+lastIndex, std::back_inserter(DigitsInROF));
+
+    // fill in the message
+    //   LOG(INFO) << " runDigitsRootFileReaderMFT::run. In this ROF there are  " << DigitsInROF.size() << " digits";
+    pc.outputs().snapshot(Output{"MFT", "DIGITS", 0, Lifetime::Timeframe}, DigitsInROF);
+    pc.outputs().snapshot(Output{"MFT", "MFTDigitROF", 0, Lifetime::Timeframe}, oneROFvec);
+  }
+
+private:
+  std::unique_ptr<TFile> mFile = nullptr;
+  unsigned long currentROF = 0;
+  
+}; // end class definition
+
+
+// clang-format off
```

Comment:
We use this if clang-format really breaks the formatting of complex structures. Please try to see if it breaks it in your case if you don't use `clang-format off`

Reply:
Removed clang-format off.